lib/Recur/RRuleIterator.php

```diff
@@ -588,6 +588,11 @@ protected function nextYearly() {
                     (int)$currentDayOfMonth
                 );
 
+                // maximum year 10000 should be safe to stop (2016 + max 3500 yearly occurrences)
+                if ((int)$currentYear > 10000) {
+                    throw new NoInstancesException('This recurrence rule does not generate any valid instances, stopped in year 10000');
```

Comment:
Any new ideas here? The endless loop is still present, it's a potential DoS attack vector. If running in a webserver, the request takes 100% CPU and high memory. It's killed hopefully after max 30 seconds, but 10 or 20 parallel requests that trigger this endless loop is a serious problem.
If using the VObject parser in a Cronjob or some other CLI script (like the Migration script), without a script timeout limit, it will never end.

We should fix this problem. Please tell me where to catch the Exception and how to change the valid() function, or maybe do it yourself directly, might be easier ;-)